### PR TITLE
fix: use correct rust-analyzer download URL on Windows

### LINKED ISSUE #144: Update URL and zip extraction to support Windows
Heya, currently on Windows, the plugin attempts to download this:

```
https://github.com/rust-analyzer/rust-analyzer/releases/download/2024-11-25/rust-analyzer-x86_64-pc-windows-msvc.gz
```

whereas it's published as `zip` (see [2024-11-25](https://github.com/rust-lang/rust-analyzer/releases/tag/2024-11-25)), and has been for a while.

This results in the "HTTP Error 404: Not Found" error as seen in https://github.com/sublimelsp/LSP-rust-analyzer/issues/140.

I've got a PR incoming to add support.

### PR DESCRIPTION
**Pull request by azriel91**
* Closes #144.

This downloads and extracts `rust-analyzer.exe` to the `C:\Users\%user%\AppData\Local\Sublime Text\Package Storage\LSP-rust-analyzer` folder:

image

### CODE REVIEW

File: plugin.py [hunk 1]
```diff
@@ -158,18 +159,25 @@ def install_or_update(cls) -> None:
                 shutil.rmtree(cls.basedir())
             os.makedirs(cls.basedir(), exist_ok=True)
             version = cls.server_version()
-            url = URL.format(tag=TAG, arch=arch(), platform=platform())
-            gzipfile = os.path.join(cls.basedir(), "rust-analyzer.gz")
+            is_windows = sublime.platform() == "windows"
+            extension = "zip" if is_windows else "gz"
+            url = URL.format(tag=TAG, arch=arch(), platform=platform(), ext=extension)
+            gzipfile = os.path.join(cls.basedir(), f"rust-analyzer.{extension}")
```
Comment (rchl) [inline plugin.py:165]: lets call it `archive_file`

File: plugin.py [hunk 2]
```diff
@@ -158,18 +159,25 @@ def install_or_update(cls) -> None:
...
             with urllib.request.urlopen(url) as fp:
                 with open(gzipfile, "wb") as f:
                     f.write(fp.read())
-            with gzip.open(gzipfile, "rb") as fp:
-                with open(serverfile, "wb") as f:
-                    f.write(fp.read())
+
+            if is_windows:
+                with zipfile.ZipFile(gzipfile, "r") as zip_ref:
+                    zip_ref.extract("rust-analyzer.exe", cls.basedir())
```
Comment (rchl) [inline plugin.py:176]: Can use the `gzipfile`/`archive_file` variable here.

Comment (rchl) [reply plugin.py:176]: Sorry, got confused. But we could extract the `"rust-analyzer.exe" if is_windows else "rust-analyzer"` line into its own variable and reuse here.

Comment (azriel91) [reply plugin.py:176]: ah yes, I was wondering if I should do that. One moment

**Review by rchl — APPROVED** [review]

**Comment by rchl** [comment]
Thanks

**Pull request merged**